core/src/main/kotlin/com/karumi/kotlinsnapshot/KotlinSnapshot.kt

```diff
@@ -1,17 +1,28 @@
 package com.karumi.kotlinsnapshot
 
 import com.karumi.kotlinsnapshot.core.Camera
+import com.karumi.kotlinsnapshot.core.SerializationModule
+import com.karumi.kotlinsnapshot.core.KotlinSerialization
 
-class KotlinSnapshot(snapshotsFolder: String = "") {
+class KotlinSnapshot<in A>(
```

Comment:
I'm trying to figure out how is the generic working here if we are only using `Any` at the end of the chain (`KotlinSerialization : SerializationModule<Any>`). Do you mind describing why is this necessary or if it's meant to be extended in some way in the future?

Reply:
I think I understand now, if we want to implement a Room extension we'd create:

```kotlin
class RoomSerialization : SerializationModule<Room> {
    fun serialize(value: Room?): String {...}
}

private val roomCamera = Camera(RoomSerialization())
fun Room.matchWithSnapshot(snapshotName: String? = null) {
    roomCamera.matchWithSnapshot(snapshotName, this)
}
```

Great job @tonilopezmr

Reply:
Include add a

```
fun <reified A> A.matchWithSnapshot(snapshotName: String? = null)  {
      val serializer = serializerFactory(it)
      val camera = Camera(serializer)
      camera.matchWithSnapshot(snapshotName, this)
}
```

under the hood, if the user has registered (or us) the serializer that matches with the type... then magic happens